src/NServiceBus.RabbitMQ.AcceptanceTests/ConfigureEndpointRabbitMQTransport.cs

```diff
@@ -76,13 +70,31 @@ ConnectionFactory CreateConnectionFactory()
     {
         var connectionFactory = new ConnectionFactory
         {
-            UserName = connectionStringBuilder["username"].ToString(),
-            Password = connectionStringBuilder["password"].ToString(),
-            VirtualHost = connectionStringBuilder["virtualhost"].ToString(),
-            HostName = connectionStringBuilder["host"].ToString(),
             AutomaticRecoveryEnabled = true
         };
 
+        object value;
+
+        if (connectionStringBuilder.TryGetValue("username", out value))
+        {
+            connectionFactory.UserName = value?.ToString();
```

Comment:
The reason it works right now is that [RabbitMQ.Client.ConnectionFactory has the same defaults](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/63aeeb8d0b7908e3dcbd87905b5f37515ed6e2f6/projects/client/RabbitMQ.Client/src/client/api/ConnectionFactory.cs#L126-L142).

This begs the question of whether we are unnecessarily repeating those defaults in `ConnectionConfiguration`.

One could argue that it insulates us from RabbitMQ.Client changing their defaults, but I'd argue that that would be such an enormous breaking change for the community that they'll never do it, and we also watch their releases with a close eye so if the ever did it, I think we'd spot the problem very early.